Serial support

flip.count is not longer known, changed for clean compile.
This new API is not tested.
(Anders and Henry)

Index: linux-2.6.17-source/drivers/char/Makefile
===================================================================
--- linux-2.6.17-source.orig/drivers/char/Makefile
+++ linux-2.6.17-source/drivers/char/Makefile
@@ -25,6 +25,7 @@
 obj-$(CONFIG_CYCLADES)		+= cyclades.o
 obj-$(CONFIG_STALLION)		+= stallion.o
 obj-$(CONFIG_ISTALLION)		+= istallion.o
+obj-$(CONFIG_COOPERATIVE)     	+= cocd.o
 obj-$(CONFIG_DIGIEPCA)		+= epca.o
 obj-$(CONFIG_SPECIALIX)		+= specialix.o
 obj-$(CONFIG_MOXA_INTELLIO)	+= moxa.o
Index: linux-2.6.17-source/drivers/char/cocd.c
===================================================================
--- linux-2.6.17-source.orig/drivers/char/cocd.c
+++ linux-2.6.17-source/drivers/char/cocd.c
@@ -93,13 +93,9 @@
 				return;
 			}
 			up(&cocd->sem);
-			if(e < (m += (TTY_FLIPBUF_SIZE - tty->flip.count)))
-				m = e;
-			while(p < m)
-				tty_insert_flip_char(tty, *(p++), 0);
-			if(tty->flip.count >= TTY_FLIPBUF_SIZE) {
-				tty_flip_buffer_push(tty);
-			}
+			m += tty_buffer_request_room(tty, e - m);
+			p += tty_insert_flip_string(tty, p, m - p);
+			tty_flip_buffer_push(tty);
 			down(&cocd->sem);
 		}
 		list_del(&input->node);
@@ -122,8 +118,7 @@
 	}
 	up(&cocd->sem);
 
-	if(tty->flip.count)
-		tty_flip_buffer_push(tty);
+	tty_flip_buffer_push(tty);
 }
 
 static int cocd_open(struct tty_struct *tty, struct file * filp)
